cmd/vic-machine/configure/configure.go

```diff
@@ -201,6 +278,83 @@ func updateSessionEnv(sess *executor.SessionConfig, envName, envValue string) {
 	sess.Cmd.Env = newEnvs
 }
 
+func (c *Configure) processCertificates(conf *config.VirtualContainerHostConfigSpec) error {
+
+	if !c.regenerateCerts && (c.skey == "" || c.scert == "") {
+		log.Infof("No certificate regeneration requested. No new certificates provided. Certificates left unchanged.")
+		return nil
+	}
+
+	if c.certPath == "" {
+		c.certPath = c.DisplayName
+	}
+
+	_, err := os.Lstat(c.certPath)
+	if err == nil || os.IsExist(err) {
+		return fmt.Errorf("Specified or default certificate output location \"%s\" already exists. Specify a location that does not yet exist with --cert-path to continue or do not specify --regenerate-certs if, instead, you want to load certificates from %s", c.certPath, c.certPath)
+	}
+
+	var debug int
+	if c.Debug.Debug == nil {
+		debug = 0
+	} else {
+		debug = *c.Debug.Debug
+	}
+
+	// prepare yourself for some mighty descriptive names and struct copying
+	seed := &common.CertSeed{
```

Comment:
Definitely took that approach at first. Got a lot of import cycles due to the package hierarchy. Used this uglier but more straightforward approach to avoid import cycles. I'll see, now that I have everything working, if it's possible for me to use composition here.